kong/router/fields.lua

```diff
@@ -288,8 +287,8 @@ local function get_cache_key(fields, params, ctx)
 end
 
 
-local function get_atc_context(schema, fields, params)
-  local c = context.new(schema)
+local function get_atc_context(context, fields, params)
+  context:reset()
```

Comment:
This function should not be called "get_atc_context" anymore as it is no longer allocating. `fill_atc_context` will be better.

Also, we should call the `context:reset()` method outside of `fill_atc_context`, inside `matching`, as `matching` is responsible for providing or resetting the context, this function doesn't care if it is a reused context or new context. Otherwise this function is too coupled with the `matching` function and should just be moved there instead.

Reply:
done